torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -80,7 +80,19 @@ bool FunctionParameter::check(PyObject* obj) {
     case ParameterType::TENSOR: {
       return THPVariable_Check(obj);
     }
-    case ParameterType::SCALAR: return THPUtils_checkDouble(obj);
+    case ParameterType::SCALAR: {
+      // NOTE: we don't currently accept most NumPy types as Scalars. np.float64
+      // is okay because it's a subclass of PyFloat. We may want to change this
+      // in the future.
+      if (THPUtils_checkDouble(obj)) {
+        return true;
+      }
```

Comment:
shouldn't we also have a checkLong here?

Reply:
checkDouble allows PyInt and PyLong